pkg/comp-functions/runtime/function_mgr.go

```diff
@@ -178,6 +193,45 @@ func (m Manager) RunFunction(ctx context.Context, req *fnv1beta1.RunFunctionRequ
 	return sr.GetResponse()
 }
 
+func (m *Manager) extractSteps(service any) []any {
+	// We get the first and only field of the service struct
+	v := reflect.ValueOf(service).Field(0)
+	values := make([]any, v.Len())
+
+	for i := 0; i < v.Len(); i++ {
+		values[i] = v.Index(i).Interface()
+	}
+	return values
+}
+
+func (m *Manager) getStepName(step any) string {
+	v := reflect.ValueOf(step).Field(0)
+	if v.Kind() != reflect.String {
+		panic(fmt.Sprintf("%s not a string, can't get step name", v.Kind()))
+	}
+
+	return v.String()
+}
+
```

Comment:
Can you please add here the reason as to why we use reflection. Or maybe in any other function you see fit.

Reply:
Done.